Sources/Client/Model/Channel+Requests.swift

```diff
@@ -0,0 +1,457 @@
+        guard changed else {
+            completion(.success(.init(channel: self)))
+            return .empty
+        }
+        
+        return Client.shared.request(endpoint: .updateChannel(.init(data: .init(self))), completion)
+    }
+    
+    /// Delete the channel.
+    /// - Parameter completion: a completion block with `Channel`.
+    @discardableResult
+    func delete(_ completion: @escaping Client.Completion<Channel>) -> URLSessionTask {
+        return Client.shared.request(endpoint: .deleteChannel(self)) { (result: Result<ChannelDeletedResponse, ClientError>) in
+            completion(result.map({ $0.channel }))
+        }
+    }
+    
+    // MARK: - Message
+    
+    /// Send a new message or update with a given `message.id`.
+    /// - Parameters:
+    ///   - message: a message.
+    ///   - completion: a completion block with `MessageResponse`.
+    @discardableResult
+    func send(message: Message, _ completion: @escaping Client.Completion<MessageResponse>) -> URLSessionTask {
+        if isActive {
+            return sendForActiveChannel(message: message, completion)
+        }
+        
+        return query { [weak self] _ in self?.sendForActiveChannel(message: message, completion) }
+    }
+    
+    @discardableResult
+    private func sendForActiveChannel(message: Message,
+                                      _ completion: @escaping Client.Completion<MessageResponse>) -> URLSessionTask {
+        let completion = doAfter(completion) { response in
+            if response.message.isBan {
+                if let currentUser = User.current, !currentUser.isBanned {
+                    var user = currentUser
+                    user.isBanned = true
+                    Client.shared.user = user
+                }
+            }
+        }
+        
+        return Client.shared.request(endpoint: .sendMessage(message, self), completion)
+    }
+    
+    /// Send a message action for a given ephemeral message.
+    /// - Parameters:
+    ///   - action: an action, e.g. send, shuffle.
+    ///   - ephemeralMessage: an ephemeral message.
+    ///   - completion: a completion block with `MessageResponse`.
+    @discardableResult
+    func send(action: Attachment.Action,
+              for ephemeralMessage: Message,
+              _ completion: @escaping Client.Completion<MessageResponse>) -> URLSessionTask {
+        let endpoint = Endpoint.sendMessageAction(.init(channel: self, message: ephemeralMessage, action: action))
+        return Client.shared.request(endpoint: endpoint, completion)
+    }
+    
+    /// Mark messages in the channel as read.
+    /// - Parameter completion: a completion block with `Event`.
+    @discardableResult
+    func markRead(_ completion: @escaping Client.Completion<Event>) -> URLSessionTask {
+        guard config.readEventsEnabled else {
+            return .empty
+        }
+        
+        Client.shared.logger?.log("🎫 Send Message Read. For a new message of the current user.")
```

Comment:
Why don't we inject a logger in `Channel` so we can use it everywhere within `Channel`?